pkg/operator/resource/resourceapply/core.go

```diff
@@ -79,12 +79,15 @@ func ApplyService(client coreclientv1.ServicesGetter, recorder events.Recorder,
 		typeSame = true
 	}
 
-	if selectorSame && typeSame && !*modified {
+	portsSame := equality.Semantic.DeepEqual(existingCopy.Spec.Ports, required.Spec.Ports)
```

Comment:
This can fail when `ServiceAppProtocol` feature gate is off.  In 1.18, it is off by default.

Reply:
> This can fail when `ServiceAppProtocol` feature gate is off. In 1.18, it is off by default.

If someone is setting the `AppProtocol`, I think you can just fail to apply in general for now.

Reply:
So I was testing to update the port protocol field in a service and that was done without any issue (on latest cluster - `4.5.0-0.ci-2020-05-04-102311`)

Reply:
Also tested the change manually changes in our repo and it was working, without erroring our